whole-repo-tests/test_pyright.py

```diff
@@ -173,6 +173,26 @@ def test_numpy_arrays_strategy(tmp_path: Path):
     assert errors == []
 
 
+def test_pandas_column(tmp_path: Path):
+    file = tmp_path / "test.py"
+    file.write_text(
+        textwrap.dedent(
+            """
+            from hypothesis.extra.pandas import column
+            from hypothesis.strategies import floats
+
+            x = column(name="test", elements=floats(), fill=None, unique=True)
+            y = column(name="test", dtype=float, fill=None, unique=True)
+            """
+        ),
+        encoding="utf-8",
+    )
+    _write_config(tmp_path, {"typeCheckingMode": "strict"})
```

Comment:
```suggestion
    _write_config(tmp_path, {"typeCheckingMode": "standard"})
```

Reply:
You could also move this in the test `test_revealed_types` to assert that the revealed type is the expected one.

Reply:
I disagree here: we should test with strict-mode, because we want our annotations to work for users who use strict mode.

I think we can make this work by adding an `__init__` method to the `column` class, with overloads on init to return `-> column[Any]` in the case that neither `elements` nor `fill` is passed.  It'll be fiddly, but I think useful!

Reply:
Would [pep696](https://peps.python.org/pep-0696/) be relevant here?